app/src/main/java/com/example/travelapp/Dijkstras.java

```diff
@@ -0,0 +1,131 @@
+package com.example.travelapp;
+
+import android.util.Log;
+
+import com.example.travelapp.BinaryMinHeap;
+import com.example.travelapp.fragments.Graph;
+import com.google.maps.DirectionsApiRequest;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PendingResult;
+import com.google.maps.model.DirectionsResult;
+import com.google.maps.model.Distance;
+import com.parse.ParseQuery;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.example.travelapp.fragments.ComposeFragment.API_KEY;
+
+public class Dijkstras {
+
+    private static final String TAG = "Dijkstras";
+
+    // TODO - check max number of google maps api calls
+    public static Graph createGraph(List<String> ids, Itinerary itinerary) throws InterruptedException {
+        GeoApiContext mGeoApiContext = new GeoApiContext.Builder()
+                .apiKey(API_KEY)
+                .build();
+        int totalDistance = 0;
+        Graph graph = new Graph(ids.size());
+        // calculating totalDistance
+        for (int i = 0; i < ids.size(); i++) {
+            for (int j = i + 1; j < ids.size(); j++) {
+                // get distance of directions from i to j in meters and add to total distance
+                int currentDistanceIJ =
+                        (int) itinerary.getDistance(ids.get(i), ids.get(j), mGeoApiContext).inMeters;
+                totalDistance += currentDistanceIJ;
+                // get distance of directions from j to i in meters and add to total distance
+                int currentDistanceJI =
+                        (int) itinerary.getDistance(ids.get(j), ids.get(i), mGeoApiContext).inMeters;
+                totalDistance += currentDistanceJI;
+            }
+        }
+        int interval = totalDistance / (ids.size() * 2);
+        for (int i = 0; i < ids.size(); i++) {
+            for (int j = i + 1; j < ids.size(); j++) {
+                if ((i == 0 && j == 1) || (i == 1 && j == 0)) {
+                    continue;
+                }
```

Comment:
have it break on certain conditions instead of continue on opposite conditions is less confusing in my opinion